source/conventions.md

```diff
@@ -171,23 +110,140 @@ lattice expansion where there are [`Fork`](#s:forking) elements and for evaluati
 
 Lattice elements can also be referred to by the index in which they appear in a branch
-with the first element having index one, etc. Branches do not get an index since the
+with the first element having index one, etc. For example, `"B1>>7"` matches the seventh
+element in branch `B1`.
+
+Branches do not get an index since the
 PALS standard does not mandate that the branches of a lattice be stored in an array (it
 could, for example, be a linked list).
 
+If there are multiple [`Lattice`](#s:lattice.construct) constructs, the element name may be qualified using the lattice
+name with `">>>"` as a separator. There are several permutations where `>>` and `>>>` are used:
+```{code} yaml
+{lattice-name}>>>{branch-name}>>{element-name>}
+{lattice-name}>>>{element-name}
+{branch-name}>>{element-name}
+```
+
+Regular expressions can be used. 
+Regular expressions must conform to the [PCRE2](https://www.pcre.org/) standard. 
+Regex matching is applied to the lattice name, branch name, and element name separately and
+a match to the string requires all the individual names to match.
+When applying regex to a lattice name, any prefix (anything before and including a `"::"`) and
+any suffix (anything after and including a `"#"` character) is not included in the regex match.
+For example, `"B.4>>Quadrupole::Qaf.*"` would match to all Quadrupole elements in branches 
+which have three characters
+beginning in "B" and ending in "4" with the element name beginning with "Qaf". And 
+`"B.4>>Quadrupole::Qaf.*#2"` would match to the second element matched to.
+
+Elements can be matched using a range construct which has the form
+```{code} yaml
+<ele1>:<ele2>
+```
+where `<ele1>` marks the beginning of the range and `<ele2>` marks the end of the range.
+Example:
+```{code} yaml
+Q1:Q2
+```
+In this example, the range matches all elements from `Q1` to `Q2` inclusive of `Q1` and `Q2`.
+If `<ele2>` comes before `<ele1>` the range "wraps around" the branch or beamline.
+For example, if `Q2` comes before `Q1` in the above example, the range matches all elements from
+`Q1` to the end of the line plus all elements from the beginning of the line to `Q2`.
+
+Commas `,` can be used to form the union of element sets. The syntax is
+```{code} yaml
+<element-set1>, <element-set2>, ... , <element-setN>
+```
+where `<element-set1>`, ... `<element-setN>` are element sets. 
+Example:
+```{code} yaml
+A, B, Q.*
+```
+This will match to all elements named `A`, `B`, and all elements whose name begins with `Q`.
+
+Ampersands `&` can be used to form the intersection of element sets. The syntax is
+```{code} yaml
+<element-set1> & <element-set2> & ... & <element-setN>
+```
+where `<element-set1>`, ... `<element-setN> are element sets. 
+Example:
+```{code} yaml
+Marker::.* & Q1:Q2
+```
+This will match to all `Marker` elements that are in the range from `Q1` to `Q2`.
+
+Order of presidence:
+```{code} yaml
+>>>     # Highest
+>>
+::      
+#
+:
+,
+&      # Lowest
+```
```

Comment:
General question: Is this arbitrary or did it come from somewhere? I'm asking about both the characters and their order.

Reply:
Adapted from Bmad.